basic functionality + tests (#1)
gagoar/codeowners-generator: README.md

- [Installation](#installation)
- [Usage](#usage)
- [Contributing](#contributing)
- [License](#license)

<!-- ABOUT THE PROJECT -->

## About The Project

[CODEOWNERS](https://docs.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners) are automatically requested for review when someone opens a pull request that modifies code that they own. This is a great feature, but when working on monorepos ownership is shared between teams and it becomes difficult to maintain.

`codeowners-generator` allows you to position CODEOWNERS files anywhere in your project tree and it will take care of compiling all the files into a single generated file, that Github can understand. It also can read the maintainers field in `package.json` (`useMaintainers` option in the cli ) making easy to keep CODEOWNERS and package.json in sync.

### Built With

- [ora](https://github.com/sindresorhus/ora)
- [commander](https://github.com/tj/commander.js/)
- [cosmiconfig](https://github.com/davidtheclark/cosmiconfig)

<!-- GETTING STARTED -->

### Installation

If you wish to use `codeowners-generator` as a standalone utility:

```sh
npm -g install codeowners-generator
```

> This will make the `alo` command available in your terminal.

Reviewer: was this meant to be `alo`?
```suggestion
This will make the `codeowners-generator` command available in your terminal.
```

```sh
codeowners-generator --help
```

If instead you would like to add it to a package:

```sh
npm install --only=dev codeowners-generator
```

> or

Reviewer: was this meant to be here? or perhaps you're missing a section after this?

Author: oh yeah, I was doing yarn as well, and then I gave up.

<!-- USAGE EXAMPLES -->

## Usage

Every command accepts several options through command line or custom configuration [see configuration for more](#configuration)

### Generate CODEOWNERS file

```sh

>   codeowners-generator generate

Reviewer: ```suggestion
codeowners-generator generate
```

```

> ### Generate CODEOWNERS file (using maintainers field from package.json's)

Reviewer: ```suggestion
### Generate CODEOWNERS file (using `maintainers` field from `package.json`)
```

```sh

> 
>   codeowners-generator generate --useMaintainers

Reviewer: ```suggestion
codeowners-generator generate --useMaintainers
```

```

> ### Specify CODEOWNERS (in case are named differently)

Reviewer: ```suggestion
### Specify where to look for CODEOWNERS
```

```sh

>   codeowners-generator generate --includes '**/CODEOWNERS'

Reviewer: ```suggestion
codeowners-generator generate --includes '**/my_custom/path_to/**/Codeowners.txt'
```

```

<!-- CONFIGURATION -->

## Configuration

You can configure `codeowners-generator` from several places:

### CLI options

- **includes** (`--includes`): The glob used to find CODEOWNERS files in the repo `default: **/CODEOWNERS`a

> - **useMaintainers** (`--useMaintainers`): It will use `maintainers` field from package.json to generate codeowners `default: **/package.json`

Reviewer: Would be nice to see an example of what is accepted in the `maintainers` field in `package.json`.

Is it an array of GitHub handles? Emails? Full names?
